pydantic_ai_slim/pydantic_ai/_instrumentation.py

```
@@ -65,6 +66,31 @@ class CostCalculationFailedWarning(Warning):
+    """
+    if not response.model_name:
+        # Without a model name (e.g. a synthetic response from a capability) there's nothing to price.
+        return Decimal(0)
+    try:
+        return response.cost().total_price
+    except (LookupError, ValueError):
+        # NOTE(Marcelo): We can allow some kind of hook on the provider level, which we could retrieve via
+        # `ctx.deps.model.provider.calculate_cost`, but I'm not sure how would the API look like. Maybe a new parameter
+        # on the `Provider` classes, that parameter would be a callable that receives the same parameters as `genai_prices`.
+        return Decimal(0)
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)